src/ccutil/tprintf.cpp

```diff
@@ -31,44 +31,49 @@
 
 namespace tesseract {
 
-#define MAX_MSG_LEN 2048
-
 INT_VAR(log_level, INT_MAX, "Logging level");
 
 static STRING_VAR(debug_file, "", "File to send tprintf output to");
 
-// Trace printf
-void tprintf(const char *format, ...) {
-  const char *debug_file_name = debug_file.c_str();
-  static FILE *debugfp = nullptr; // debug file
-
-  if (debug_file_name == nullptr) {
-    // This should not happen.
-    return;
-  }
+// File for debug output.
+static FILE *debugfp;
 
+// Set output for log messages.
+// The output is written to stderr if debug_file is empty.
+// Otherwise it is written to debug_file.
+// It is possible to switch between stderr and debug_file output:
+// tprintf("write to configured output\n");
+// debug_file = "";
+// tprintf("write to stderr\n");
+// debug_file = "/tmp/log";
+// tprintf("write to /tmp/log\n");
+// debug_file = "";
+// tprintf("write to stderr\n");
```

Comment:
I wonder whether such use cases exist. We could simplify the code further if we only allow setting the desired output once before the first use of `tprintf`.